addons/l10n_id_pos/models/pos_payment_method.py

```diff
@@ -0,0 +1,21 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.exceptions import UserError
+from odoo import _, models
+
+
+class PosPaymentMethod(models.Model):
+    _inherit = "pos.payment.method"
+
+    def l10n_id_verify_qris_status(self, trx_uuid):
+        """ Verify qris payment status from the provided transaction UUID
+
+        For all qris_invoice_details linked to the transaction, check the payment status
+        """
+        if self.payment_method_type != 'qr_code' or self.qr_code_method != 'id_qr':
+            return True
+        trx = self.env['l10n_id.qris.transaction'].search([('model', '=', 'pos.order'), ('model_id', '=', trx_uuid)])
```

Comment:
```suggestion
        trx = self.env['l10n_id.qris.transaction']._get_latest_transaction('pos.order', trx_uuid)
```
What if there are more than one, for any reason?